src/components/tx/SignOrExecuteForm/SafeNetTxChecks.tsx

```diff
@@ -0,0 +1,29 @@
+import { type ReactElement, useContext } from 'react'
+import { SafeNetTxSimulation } from '@/components/tx/security/safenet'
+import TxCard from '@/components/tx-flow/common/TxCard'
+import { SafeTxContext } from '@/components/tx-flow/SafeTxProvider'
+import useIsSafeNetEnabled from '@/hooks/useIsSafeNetEnabled'
+import { Typography } from '@mui/material'
+import useChainId from '@/hooks/useChainId'
+import useSafeAddress from '@/hooks/useSafeAddress'
+
+const SafeNetTxChecks = (): ReactElement | null => {
+  const safe = useSafeAddress()
+  const chainId = useChainId()
+  const { safeTx } = useContext(SafeTxContext)
+  const isSafeNetEnabled = useIsSafeNetEnabled()
```

Comment:
I am not a big fan of using hooks to retrieve the data accessible in the parent component. It gets rather hard to follow where the data is coming from. Can we pass them as a prop?

Reply:
Ah, I see what's up. You followed the example of an existing component. Nevermind then.

Reply:
I made the inner component `SafeNetTxSimulation` work on passed in props instead of reading state with hooks for this reason.